### ISSUE TITLE
Feature Request: Support seed parameter in /v1/audio/speech (TTS)

### ISSUE THREAD

**Issue by BeHappy0o0o0o0** [jundot/omlx#3297]
## Summary

The OpenAI-compatible `/v1/audio/speech` endpoint (Text-to-Speech) does not
accept a `seed` parameter. As a result, synthesizing the same text twice
produces different audio every time (random sampling), making output
non-reproducible. `/v1/chat/completions` already honors `seed` (the scheduler
calls `mx.random.seed(request.sampling_params.seed)`), so extending the same
semantics to TTS would make the API fully consistent with OpenAI's
documented `seed` behavior for audio speech.

## Environment

- oMLX version: 0.6.3
- Hardware: Apple M1 Max (arm64)
- Model: `Qwen3-TTS-12Hz-1.7B-Base` (also reproducible with `VoxCPM2`)

## Current behavior

`AudioSpeechRequest` (`omlx/api/audio_models.py`) has no `seed` field:

```python
class AudioSpeechRequest(BaseModel):
    model: str
    input: str
    voice: Optional[str] = None
    language: Optional[str] = None
    instructions: Optional[str] = None
    speed: Optional[float] = 1.0
    response_format: Optional[str] = "wav"
    ref_audio: Optional[str] = None
    ref_text: Optional[str] = None
    temperature: Optional[float] = None
    top_k: Optional[int] = None
    top_p: Optional[float] = None
    repetition_penalty: Optional[float] = None
    max_tokens: Optional[int] = None
    stream: Optional[bool] = False
    streaming_interval: Optional[float] = None
```

Because Pydantic models ignore unknown fields by default, sending `"seed": 42`
in the request body is silently discarded and never reaches the TTS engine.

The TTS path (`omlx/engine/tts.py` -> mlx-audio `qwen3_tts.Model.generate()`)
samples with `categorical_sampling(logits, temperature)` without ever setting
the MLX random state, so every request draws fresh random samples.

## Reproduction

```bash
# same text, no seed → byte-identical? No: different hash every time
for i in 1 2; do
  curl -s http://127.0.0.1:8000/v1/audio/speech \
    -H 'Content-Type: application/json' \
    -d '{"model":"Qwen3-TTS-12Hz-1.7B-Base","input":"Hello, reproducibility test.","response_format":"wav"}' \
    | shasum -a 256
done
# different hashes

# with seed → still different (seed silently ignored)
for i in 1 2; do
  curl -s http://127.0.0.1:8000/v1/audio/speech \
    -H 'Content-Type: application/json' \
    -d '{"model":"Qwen3-TTS-12Hz-1.7B-Base","input":"Hello, reproducibility test.","response_format":"wav","seed":42}' \
    | shasum -a 256
done
# different hashes — seed is dropped
```

## Expected behavior

Following OpenAI's `seed` semantics for audio:

- Same text + same `seed` (and same sampling params) → identical audio output.
- Different `seed` → different (but still reproducible per seed) output.
- `seed=None` → current random behavior (unchanged).

## Suggested implementation

1. Add `seed: Optional[int] = None` to `AudioSpeechRequest`
   (`omlx/api/audio_models.py`).
2. In `AudioSpeechRequest -> TTSEngine.synthesize()` (or directly in
   `omlx/engine/tts.py`), call `mx.random.seed(seed)` right before
   `model.generate(...)` when `seed is not None` — mirroring what the LLM
   scheduler already does at `omlx/scheduler.py`:
   ```python
   if request.sampling_params.seed is not None:
       mx.random.seed(request.sampling_params.seed)
   ```

We verified locally that seeding the MLX RNG before `Qwen3-TTS` generation is
sufficient for full determinism:

| scenario | result |
|---|---|
| no seed, temp 0.9, two runs | different hashes (random) |
| `mx.random.seed(42)`, temp 0.9, two runs | **identical hashes** |
| `seed=42` vs `seed=99` | different hashes (reproducible per seed) |
| temp 0.0 (greedy), two runs | identical hashes (already deterministic) |

## Rationale

TTS reproducibility matters for:
- Regression / A/B testing of synthesis quality.
- Caching downstream (identical input should hit the cache).
- Product scenarios where a fixed voice sample must be replayed consistently
  (like `seed` in OpenAI audio-capable LLMs / the official `audio/speech`
  ecosystem).

This would not change default behavior for existing callers (`seed` is
optional), and it aligns the TTS endpoint with the `seed` support already
present in `/v1/chat/completions`.
